hermes/src/store/storage/local_storage.rs

```diff
@@ -155,12 +157,507 @@ impl Storage for LocalStorage {
     }
 
     async fn store_accumulator_state(&self, state: super::AccumulatorState) -> Result<()> {
-        let key = state.slot;
-        self.accumulator_cache.insert(key, state);
+        let mut accumulator_cache = self.accumulator_cache.write().await;
+        accumulator_cache.push_back(state);
+
+        let mut i = accumulator_cache.len().saturating_sub(1);
+        while i > 0 && accumulator_cache[i - 1].slot > accumulator_cache[i].slot {
+            accumulator_cache.swap(i - 1, i);
+            i -= 1;
+        }
+
+        if accumulator_cache.len() > self.cache_size as usize {
```

Comment:
maybe make this a `while` ? I think `if` is also correct because the size of the deque can only grow by 1 here, but `while` would be more robust to code changes around it (like inserting multiple values at the same time)

Reply:
Makes sense